File: app/src/main/java/com/agotakiss/androidtest/domain/MovieRepository.java
package com.agotakiss.androidtest.domain;

import com.agotakiss.androidtest.data.models.Movie;

import java.util.List;

import io.reactivex.Single;

public interface MovieRepository {

    Single<List<Movie>> getPopularMovies(int page);

    Single<List<Movie>> getSimilarMovies(String movieId, int page);
}

File: app/src/main/java/com/agotakiss/movie4u/domain/repository/MovieRepository.kt
package com.agotakiss.movie4u.domain.repository

import com.agotakiss.movie4u.domain.models.Movie
import com.agotakiss.movie4u.domain.paging.PagedResult
import io.reactivex.Completable
import io.reactivex.Single

interface MovieRepository {

    fun getPopularMovies(page: Int): Single<PagedResult<Movie>>

    fun getSimilarMovies(movieId: Int, page: Int): Single<PagedResult<Movie>>

    fun getActorsMovies(actorId: Int): Single<List<Movie>>

    fun getMovieDetails(movieId: Int): Single<Movie>

    fun addToFavoriteMovies(movie: Movie): Completable

    fun deleteFromFavoriteMovies(movieId: Int): Completable

    fun getFavoriteMovies(): Single<List<Movie>>

    fun getSearchResults(queryString: String, page: Int): Single<PagedResult<Movie>>
}
